### PR TITLE
As a Taqueria dev, I want my code to be automatically formatted so that the codebase is more consistent and adheres to the project style guide at all times

### PR DESCRIPTION
**Pull request by MichalLytek**
Superseded #582 

Depends on #719

### CODE REVIEW

**Comment by hu3man** [comment]
Looks like the branch needs a rebase after our release this morning. Will review when the merge conflicts are resolved

**Comment by hu3man** [comment]
Hey @MichalLytek, thanks for updating the branch

I'm trying to test it out and am running into the following error:
```
❯ git commit -m  "test dprint" 
✖ No valid configuration found.
husky - pre-commit hook exited with code 1 (error)
```

Am I doing something wrong?

**Comment by MichalLytek** [comment]
@hu3man Sorry, it was rebase issues - fixed that and rebased again on `main`

**Comment by mweichert** [comment]
Hi @MichalLytek,

Would you mind adjusting this PR to be limited to just the configuration please? That's going to be easier to get merged in I think. We can then activate it once the list of long-living branches is a low number.

Thanks,
Mike

**Comment by MichalLytek** [comment]
@mweichert Pushed rebased changes with just config and git commit hook 😉

File: dprint.json [hunk 1]
```diff
@@ -0,0 +1,18 @@
+{
+	"lineWidth": 120,
+	"useTabs": true,
```
Comment (MichalLytek) [inline dprint.json:3]: Everyone can adjust the view to the needs - 2 space, 4, even 8 😉 
https://www.kindacode.com/article/vs-code-how-to-change-indentation-2-spaces-4-spaces/
https://docs.github.com/en/account-and-profile/setting-up-and-managing-your-personal-account-on-github/managing-personal-account-settings/managing-your-tab-size-rendering-preference

**Comment by MichalLytek** [comment]
Second thing to discuss - I would recommend this settings in VSCode:
```
"dprint.path": "./node_modules/dprint/dprint",
"[javascript, typescript]": {
  "editor.defaultFormatter": "dprint.dprint",
  "editor.formatOnSave": true
}
```
But the json file is git ignored so I can't commit the configuration. Any other way suggested to spread the knowledge about the configuration?

**Comment by MichalLytek** [comment]
> Would you mind adjusting this PR to be limited to just the configuration please? That's going to be easier to get merged in I think. We can then activate it once the list of long-living branches is a low number.

In some point we might need to force format of the full codebase on `main` branch. Without this, every PR will have tons of ghost changes due to reformatting.

The easiest approach is to merge this PR with config, then every other branch owner pull those changes and run `npm format` on his own codebase. We then merge to `main` the reformat of the whole codebase and since then, all the branches and PRs are now compatible and easy to merge.

**Review by mweichert — APPROVED** [review]

File: dprint.json (hunk 1 above)
Comment (AlirezaHaghshenas) [reply dprint.json:3]: I'm always in favour of tabs for that exact reason. I'll ask the team to see if there is strong objection.

**Review by hu3man — APPROVED** [review]

**Pull request merged**